src/System.Private.CoreLib/shared/System/SpanHelpers.Byte.cs

```diff
@@ -959,25 +960,34 @@ public static unsafe int LastIndexOfAny(ref byte searchSpace, byte value0, byte
         public static unsafe bool SequenceEqual(ref byte first, ref byte second, nuint length)
         {
             if (Unsafe.AreSame(ref first, ref second))
+            {
                 goto Equal;
+            }
 
+            bool result = false;
             IntPtr i = (IntPtr)0; // Use IntPtr for arithmetic to avoid unnecessary 64->32->64 truncations
             IntPtr n = (IntPtr)(void*)length;
 
-            if (Vector.IsHardwareAccelerated && (byte*)n >= (byte*)Vector<byte>.Count)
+            // Conditinally jump forward for Vector versions to give the non-Vector variant used for shorter
```

Comment:
Typo: should be "Conditionally"

Reply:
Externalized the Vector portions in methods to drop the double `vzeroupper` on short lengths